### ISSUE TITLE
Publisher Stream is not created

### ISSUE THREAD

**Issue by gowtham1094** [opentok/opentok-ios-sdk-samples-swift#111]
**Issue:**
Publisher stream is not creating whenever app launches through VOIP notification and goes through the Call Kit.

We are attaching the example project in which you guys can test the failure cases and helpful to debug the issue.

**Example Project**
Example project is same as basic video chat example with few additional changes we made like integrated call kit to raise whenever there is VOIP and start connecting whenever user press the answer button. 

**Note:**
Before testing it add proper **apiKey, sessionId and token**.
Curl command for triggering VOIP placed inside in App delegate file in which youguys have to change the device token and **pem file** and **password** of it

**Steps To Reproduce:**
1)Run the app and start connect and publish in User A Phone
2)Run and kill the app in User B Phone
3)Trigger the VOIP for User B Phone
4)VOIP will receive to User B Phone and start CallKit caller screen
5)User B answer's the call
6)Start the session connection
7)Start the publish

**Publish stream is not created** , since publish stream not created for User B, User A wont get call back for stream created so User A cant subscribe to User B stream too.

**Logs whenever connecting normally**
doConnect
Session connected
sessionDidConnect
doPublish
publisher streamCreated

**Logs whenever connecting through callkit**
doConnect
Session connected
sessionDidConnect
doPublish

[Basic-Video-Chat.zip](https://github.com/opentok/opentok-ios-sdk-samples-swift/files/4399523/Basic-Video-Chat.zip)

**Comment by bsrao** [comment]
@gowtham1094 Can you disable videoTrack by setting publisher.videoTrack = false right after you create the publisher. By default the OpenTok cloud expects both audio and video but when you launch from Callkit UI, camera won't be enabled and that results streamCreated event not getting fired!

**Comment by gowtham1094** [comment]
Hi @bsrao ,
Thanks for you response, i have integrated the CallKit similar to this `https://github.com/opentok/CallKit`, which resolve the callee problem on time of answering through call kit. But I faced the same issue (publish stream not created) on caller side on very first call - after new build installed or killed the app then initiate call. After the first call it is working perfectly fine for all N number times.

After lots of try,We have added 
`OTAudioDeviceManager.setAudioDevice(OTDefaultAudioDevice.sharedInstance())`
before creating the outgoing call similar to the incoming call which fix our problem.

So is this proper fix, or we need to something else.
Also as u mentioned in your comment `streamCreated event not getting fired!` so OpenTok supposed to tell the client that something went wrong because of that publisher stream not created.
